components/client-common.rst

```diff
@@ -76,3 +76,37 @@ If one or more of the requests throw exceptions, they are added to the
     } catch (BatchException $e) {
         var_dump($e->getResult()->getExceptions());
     }
+
+
+HTTP Client Router
+------------------
+
+This client accepts pairs of clients (both sync and async) and request matchers.
+Every request is "routed" through this single client, checked against the request matchers
+and sent using the appropriate client. If there is no matching client, an exception is thrown.
+
+This allows to inject a single client into multiple services,
+but under the hood there can be clients configured for each of them::
```

Comment:
i would first say the generic thing, and then have a new sentence with some examples what this allows to do.